### PR TITLE
Airburst 2: This time is evil(ish)

### PR DESCRIPTION
**Pull request by MajManatee**
# Document the changes in your pull request
Airburst ammo is cool, so i expanded on it
This adds two new ammo types for shotguns (gasp)
the first is a non lethal, craftable, airburst round that pops in a 360 after traveling 4 tiles, dispersing 22 pellets and doing about 15 stamina each
The second is a magazine for the syndicate shotgun, containing the EXACT same ammo, but it does normal damage instead of stamina damage.

# Spriting

the FAT shells
image

the DRUM mags
image

# Wiki Documentation
as of... uploading this (it will probably change)
One shell takes 5 blackpowder, 1 plastic, and a techshell to craft.
One shell pops into 22 pellets with a variance of 360 after traveling 4 tiles (ish)
The shell does 2 (two) damage until it pops, encouraging people to close the gap between the fighter.
The pellets do 15 stamina damage each. Rubbershot shotguns do a total of 72 ish (i think). This does a total of around 60-45 if you are next to it when it pops

The syndicate version does the EXACT SAME THING but its lethal. 
The mag can be bought for 2 TC (because you are so likely to kill your team mates)

# Changelog

:cl:  
rscadd: New Airburst shotgun ammo, Syndicate, and NT versions. (lethal and non-lethal)
/:cl:

### CODE REVIEW

**Comment by Moltijoe** [comment]
yeah, so i can't review this since it's literally just my code again

love the idea though

**Comment by MajManatee** [comment]
erm no proof it was your code I (copy and pasted) wrote it

**Comment by Moltijoe** [comment]
> erm no proof it was your code I (copy and pasted) wrote it

ah fair enough (you still have the comments)

**Comment by MajManatee** [comment]
no comments saying "Molti's Code"
if thats how it worked i couldnt review most projectile code by now

**Review by cuackles — APPROVED** [review]
wide shells do be a bit odd but they do make sense cus they filled with (???) so it fits, aproved

**Comment by Mqiib** [comment]
Thicc shells

**Comment by ynot01** [comment]
the sprites are funy

**Comment by ReddicusDragon** [comment]
mmm cornflakes

**Comment by MajManatee** [comment]
well if it isnt my arch nemesis... Ammo.dmi

File: code/modules/projectiles/projectile/bullets/shotgun.dm [hunk 1]
```diff
@@ -124,6 +124,47 @@
...
+
+/obj/item/projectile/bullet/pellet/shotgun_airburst_casing
+	name = "airburst rubber casing"
+	damage = 2 // The shell for the bullets, not designed to hit people
+	range = 7 // roughly 4 tiles 
+	sharpness = SHARP_NONE
+
+/obj/item/projectile/bullet/pellet/shotgun_airburst_casing/on_range()
+	var/obj/item/ammo_casing/shotgun/airburst_pellet_shell/P = new(get_turf(src))
+	var/mob/living/L = new (get_turf(src))//it's jank, but casings can only be shot via a mob's location
```
Comment (JohnFulpWillard) [inline code/modules/projectiles/projectile/bullets/shotgun.dm:137]: WTF...?
You should make ``fire_casing`` pass by when the user arg isn't passed, not make a whole new person for every single bullet!

If you want to go the extra mile, you could even make the user arg be 'source of the firing', and check if it's a mob when necessary, so you can have the 'source' be this pellet instead.

File: code/modules/projectiles/projectile/bullets/shotgun.dm [hunk 2]
```diff
@@ -124,6 +124,47 @@
...
+	P.fire_casing(get_edge_target_turf(firer, get_dir(firer, original)), L)
```
Comment (JohnFulpWillard) [inline code/modules/projectiles/projectile/bullets/shotgun.dm:138]: Even if it isn't used currently, ``fired_from`` can and should be the arg used for admin logs to tell who shot something. It's currently done for chemical darts, but I think these bullets lack critical admin logging.

**Comment by ToasterBiome** [comment]
Reopen with more work

**Pull request closed without merging**